kettle/model.py

```diff
@@ -166,9 +166,13 @@ def test_results_for_build(self, path):
         for dataz, in self.db.execute(
                 'select data from file where path between ? and ?',
                 (path, path + '\x7F')):
-            data = zlib.decompress(dataz).decode('utf-8')
-            if data:
-                results.append(data)
+            try:
+                data = zlib.decompress(dataz).decode('utf-8', 'replace')
+                if data:
+                    results.append(data)
+            except UnicodeDecodeError:
+                print(f'Failed to decode data for {path}')
+                break
```

Comment:
do we want to break here? or continue on to next data?

Reply:
This is all data within the same job, all test results. If one part of it is corrupt it looks like it can cascade. Because the decompress takes so long and the jobs with this issue are so old, I thought it would be more efficient to give up than skip that shard of data. Down to remove it again though.

Reply:
/lgtm

Reply:
*also the `len` of data is almost always 1, AFAIK. Have not ever had a cursor object come back with more than one entry